### ISSUE TITLE
Samsung C7000 TV "Video resolution not supported" when switching from interlaced to progressive scan

### ISSUE THREAD

**Issue by rapperskull** [ikorb/gcvideo#70]
Today I received my Bitfunx HDMI adapter, connected it to my PAL GameCube, and updated it to v3.0e. I was playing around with 240p test suite and found out that my TV supports all modes except for 288p, so I disabled the Linedoubler function for all modes except this one, but I encountered a problem when switching from interlaced to progressive modes (not the other way around though).

This problem is very apparent when the GC boots, since it starts in 576i, then Swiss decides to switch to 576p. When this happens, my TV hangs on a black screen for one or two seconds, then shows a "Video resolution not supported" message. The only way to get an image is to switch to another input on the TV (for example VGA) and then back to HDMI. This happens both with a "normal" HDMI port and with the one marked as HDMI/DVI.

Both "Fix resolution" and "Fix sync timing" are on. I tried enabling "Sample rate hack" and switching to "YC444" color format, but nothing changed.

Enabling the Linedoubler option for 576i fixes this problem, but it's not the ideal solution, since the image flickers a lot compared to the interlaced signal passed directly to the TV.

I also made some tests with 240p test suite and found out the same problem occures when switching from 480i to 480p, but not the other way around.

**Comment by rapperskull** [comment]
Since this problem happens with the XRGB Mini, and we know the resolution is detected as 360x480 instead of 720x480, my guess is that the pixel repeat flag is kept set during the switch.

I have no experience with VHDL, but I find [this line of code](https://github.com/ikorb/gcvideo/blob/d5989ead04c088575e3bbd91c426084331bca71d/HDL/gcvideo_dvi/src/gcdv_decoder.vhd#L95) a little strange. Maybe this is useful, maybe not.

**Comment by ikorb** [comment]
Oh, that's a lot of edits that I didn't see because Github does not send any mails for them.

Is "Enhanced DVI mode" enabled or disabled in your configuration? The 360x480 problem of the XRGB Mini should disappear if it is enabled.

**Comment by rapperskull** [comment]
Enhanced DVI mode is enabled and the "Display as 16:9" option does nothing.

**Comment by rapperskull** [comment]
BTW, on an unrelated note, the lower 4 bytes of AVI InfoFrame Data Byte 2 should be set to 8 (0b1000), to indicate "Same as picture aspect ratio", since 0 is a reserved value. I don't think this will make a difference with the TV ignoring the aspect ratio setting, but anyway...

**Comment by rapperskull** [comment]
I did some more tests with Enhanced DVI off and I think the problem is that some info is wrongly reported.

In fact, without enhanced DVI, I can use all modes without linedoubler (even 288p).

240p is detected by the TV as 1440x240p, 480i as 1440x480i, 288p as 1440x288p and 576i as 1440x576i. 480p and 576p are of course unaffected.

It would be perfect if it wasn't for the missing audio, so my request would be, if it's not possible to fix the problem, to have an option to only enable the audio with Enhanced DVI and not the display information.

I think however that the problem is with the info that's being sent. Unfortunately my TV does not report what the unsupported resolution should be, so I can't give you more info.

**Comment by ikorb** [comment]
Thanks for the information. Shouldn't the contents of the lower 4 bits of data byte 2 be irrelevant if DB1 bit 4 is set to 0?

Usually I would try to create a test version with these changes and attach it here, but currently I cannot create new builds of GCVideo. I hope I can get back to it this weekend, but I can't make any promises yet.

**Comment by rapperskull** [comment]
Yeah, it shouldn't matter, but maybe some TVs rely on that information.

Don't worry, I can wait. In the meanwhile, I'm trying to re-create the build environment, but I see that's a lot of work.

**Comment by rapperskull** [comment]
Hi, have you managed to get access to your build environment? If not, and you have some idea you want to try, I can compile a build and test it by myself.

**Comment by ikorb** [comment]
Ok, I had a test build ready for you, but whiile playing around with it I noticed some weirdness which points to a bug that I want to squash first.

Instead I can supply you with two more questions: Do you have Output options->Display as 16:9 enabled? If so, does the problem disappear if you disable it?

**Comment by rapperskull** [comment]
Yes, I had that enabled, and no, it doesn't make any difference.

**Comment by ikorb** [comment]
Oh well, that would've been too easy...

Attached is a test build that has new options in the Advanced submenu to disable each type of info frame individually (technically: replace with a blank one...). It also changes the AVI infoframe as per your suggestions and features two other small, yet unreleased fixes, one of which I alluded to above.

If your TV really dislikes something about that infoframe, disabling it might lead to a clue. I'm not sure what the reason could be though, according to a low-end commercial analyzer box the only difference between the frames for 480i and 480p are the pixel repetition count and video mode code.

Please note: This build is not a production-ready version, "Save settings" will do nothing because I had to comment out the code to free up some space.

[updater-IFTST1-gc.zip](https://github.com/ikorb/gcvideo/files/6949743/updater-IFTST1-gc.zip)

**Comment by rapperskull** [comment]
First of all, thank you for the test build. I tested it and noticed that there's way more lag before any image is displayed on startup (the GC animation shows up very late).
I tried it without changing any option and sure enough the problem is the same. Disabling Video IF fixes it, and the resolutions are detected by the TV as reported above. Even 288p works! If I have to guess, it's that a wrong IF is sent during the switch, and the TV can't recover on its own after getting a wrong resolution. Don't know why switching to another input and back fixes the 576i to 576p and 480i to 480p switch but not 288p though.
I haven't tried all combinations, as disabling Video IF just worked, but I tried disabling everything except that one and the problem was the same.

**Comment by ikorb** [comment]
Your TV is weird... The attached version disables enhanced DVI until the next frame if the output mode changes between 15 and 31kHz, which disables all infoframes during the transition.

[updater-IFTST2-gc.zip](https://github.com/ikorb/gcvideo/files/6950902/updater-IFTST2-gc.zip)

If it does not fix the problem, it would be interesting to know if your TV stops displaying a picture with the following sequence of actions:
1. Preparation: Load something that outputs 480i or 576i on the Cube, disable linedoubling, enable enhanced mode, enable video IF, TV should show a picture
2. Disable video IF
3. Enable line doubler
4. Enable video IF

**Comment by rapperskull** [comment]
Unfortunately the build you provided does not work. I tried your steps and the TV always showed an image. In fact, when Video IF is disabled, I can change to any video mode without any issue.

**Comment by ikorb** [comment]
"does not work" as in "could not get it to install/run" or as in "did not change the symptoms"?

**Comment by rapperskull** [comment]
Didn't change anything unfortunately.

**Comment by ikorb** [comment]
Thanks. Your TV continues to confuse me. ;)

Continuing on the assumption that it dislikes something about the video infoframe, I captured the one used by an Extron scaler and tweaked GCVideo to match it (this may or may not break then 16:9 switch).

[updater-IFTST3-gc.zip](https://github.com/ikorb/gcvideo/files/6951609/updater-IFTST3-gc.zip)

If this still doesn't work I guess I'll have to bite the bullet and add the video IF option into the main branch.

**Comment by rapperskull** [comment]
Sorry, didn't see your last message. Just tested IFTST3 and it works! I tried to switch back and forth with 240p test suite and never got any problem. Unfortunately, 288p still doesn't work. Maybe that IF still needs some tweaking? Thank you for your time and patience.

BTW the 16:9 option does nothing on my TV, since it disables the Auto Width mode, so I have to manually set it to 4:3 or 16:9. Don't know under which conditions the TV disables that option, since the manual is pretty vague.

**Comment by ikorb** [comment]
There isn't much I can do about the Power-to-Image delay, if I remember correctly GCVideo produces a video signal within 250ms - but the TV needs to detect it, sync with its clock, start to decode the signals, figure out the image size, adjust its signal processing etc. Some are faster, some are slower and some seem to throw a dice internally to select how long they want to take this time. I suspect no manufacturer ever tried to optimize that process. =(

Since your new TV observations have given me a glimmer of hope that the issue can be fixed without adding new menu options(*) I'll build a version where all the differences between the infoframes can be switched individually. You get to play with these options so other people don't have to. ;)

As for 288p: You could try to enable the "Report 240p as 480i" option in the advanced menu, despite the name it also applies to 288p.

4:3/16:9 switching can be a bit weird depending on the TV - I have one that supports it if I have selected auto aspect on the TV, but it refuses to do so in 240p and 288p modes. It probably does not help that there are multiple aspect ratio indicators in the infoframe.

(*) because menu options increase code size, requiring more free RAM, which requires combing through the code to find something else that can be optimized to free up a few bytes of RAM

**Comment by rapperskull** [comment]
The lag has been introduced with IFTST1 though. With v3.0e was negligible. I'll make some more tests, maybe turning it on and off again in sequence is causing the TV to lag.
As for 288p, I will try that setting, but without Video IF it works perfectly fine, so I think the problem is again with the IF.

**Comment by rapperskull** [comment]
Just tested the "Report 240p as 480i" option, but nothing changed. The TV still reports 240p as 240p (and not 480i) and 288p as unsupported. Tested again with Video IF disabled and 288p is reported as 1440x288.

**Comment by ikorb** [comment]
The only difference between the IFs for 240p. 288p, 480i and 576i is the video mode code and turning on the "240p as 480i" option uses the 480i/576i codes for 240p/288p modes. It's really weird that your TV rejects one of those four if IFs are on, even if the IF is identical to another mode that works. I even checked via logic analyzer that the sync timing for 288p is correct and it does match the specs as well as one field of 576i as it should.

[updater-IFTST4-gc.zip](https://github.com/ikorb/gcvideo/files/6957611/updater-IFTST4-gc.zip)

IFTST4 has four options in the advanced menu that change different parts of the video IF. When they are all of, the IF is the same that is used by GCVideo 3.0e, when they are all on it is basically the same as used by an Extron scaler. I suspect that one or both of the ITContent options are the culprit. The change takes effect immediately, but your TV may need a mode switch to trigger the failure case again.

Haven't looked at the startup time thing yet.

**Comment by rapperskull** [comment]
Ok so the option that makes it work is VIF ITContent 2. With that on, and all the others off, I can switch between modes without problems.
Unfortunately, 288p does not work. I tried all 16 combinations of the four options, but none of them worked.
My speculation is that, since 288p is not officially supported, receiving the correct IF prevents the TV from displaying it, as if they tried to block it, but the underlining software can still detect 1440x288. On the other hand, also 240p is not officially supported, but works just fine, so I don't really know.
According to the user manual, the supported resolutions for HDMI, when not in PC mode, are: 1080p, 1080i, 720p, 576p, 576i, 480p and 480i. PC mode is a completely different story: it supports a lot of resolutions, but without audio over HDMI (you have to connect the analog audio jack), and no interlaced support. Still no 240p or 480p though.

My suggestion for the next release is to add "VIF ITContent 2 ON" to the default IF, and add the option to completely disable Video IF, in case someone needs it. As for 288p, I think I'll use the linedoubler option, if you ran out of ideas to try.

**Comment by ikorb** [comment]
Great, thanks for your patience testing all of this.

I'm currently working on a few other fixes, but the next release with the Complete Fix Collection should be release-ready soon(-ish). I'll try to fit a VIF switch into the remaining space, but I can't make any promises about it yet.

**Comment by rapperskull** [comment]
Hi, do you have any ETA for the new release? Thank you.

**Comment by ikorb** [comment]
It'll be released when it is done.

**Comment by ikorb** [comment]
Fix is now released as part of GCVideo 3.1
